src/coreclr/nativeaot/Runtime/DebugHeader.cpp

```diff
@@ -0,0 +1,279 @@
+    MAKE_DEBUG_FIELD_ENTRY(EEType, m_VTable);
+
+    MAKE_SIZE_ENTRY(StressLog);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, facilitiesToLog);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, levelToLog);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, totalChunk);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, logs);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, tickFrequency);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, startTimeStamp);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, startTime);
+    MAKE_DEBUG_FIELD_ENTRY(StressLog, moduleOffset);
+
+    MAKE_SIZE_ENTRY(ThreadStressLog);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, next);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, threadId);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, isDead);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, readHasWrapped);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, writeHasWrapped);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, curPtr);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, readPtr);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, chunkListHead);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, chunkListTail);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, curReadChunk);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, curWriteChunk);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, chunkListLength);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, pThread);
+    MAKE_DEBUG_FIELD_ENTRY(ThreadStressLog, origCurPtr);
+
+    MAKE_SIZE_ENTRY(StressLogChunk);
+    MAKE_DEFINE_ENTRY(StressLogChunk_ChunkSize, STRESSLOG_CHUNK_SIZE);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, prev);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, next);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, buf);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, dwSig1);
+    MAKE_DEBUG_FIELD_ENTRY(StressLogChunk, dwSig2);
+
+    MAKE_SIZE_ENTRY(StressMsg);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, fmtOffsCArgs);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, facility);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, timeStamp);
+    MAKE_DEBUG_FIELD_ENTRY(StressMsg, args);
+
+    MAKE_SIZE_ENTRY(RuntimeInstance);
+    MAKE_DEBUG_FIELD_ENTRY(RuntimeInstance, m_pThreadStore);
+
+    RuntimeInstance *g_pTheRuntimeInstance = GetRuntimeInstance();
+    MAKE_GLOBAL_ENTRY(g_pTheRuntimeInstance);
+
+    MAKE_GLOBAL_ENTRY(g_gcDacGlobals);
+
+    MAKE_GLOBAL_ENTRY(g_pFreeObjectEEType);
+
+    void *g_stressLog = &StressLog::theLog;
+    MAKE_GLOBAL_ENTRY(g_stressLog);
+
+    // Some DAC functions need to know the module base address, easiest way is with
+    // the HANDLE to our module which is the base address.
+    HANDLE moduleBaseAddress = PalGetModuleHandleFromPointer((void *)&PopulateDebugHeaders);
+    MAKE_GLOBAL_ENTRY(moduleBaseAddress);
+
+    static_assert(EEType::Flags::EETypeKindMask         == 0x0003, "SOS has a hard coded dependency on the values of EEType::Flags. If you change these values you must bump major_version_number.");
```

Comment:
Nit: I would replace 'SOS' with a more generic 'Debugging data contracts'. SOS may be the first user of this but it probably won't be the last.